firmware/lib/thermistorProbe/thermistor10k.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file thermistor10k.cpp
+ * Thermistor 10k temperature measurements.
+ */
+#include <Arduino.h>
+#include "thermistor10k.h"
+
+/**
+ * @param pin - Arduino Due pin
+ */
+Thermistor10k::Thermistor10k(uint32_t pin, double seriesResistor)
+{
+    this->thermistorPin = pin;
+    this->seriesResistorVal = seriesResistor; // measure actual resistance of the 20 kOhm series resistor
+}
+/**
+ * Obtains analog thermistor reading and converts it to temperature in Celsius
+ * @return temperature in Celsius
+ */
+double Thermistor10k::readTemperature()
+{
+    int vLevel = analogRead(thermistorPin); //defaults to 10 bit if analogReadResolution(12) is not set
+    double Tc = convertToTemp(vLevel);
+    return Tc;
+}
+
+/**
+ * Obtains analog thermistor reading and converts it to temperature in Celsius
+ * @return temperature in Celsius
+ */
+double Thermistor10k::readAveragedTemperature(int numSamples)
+{
+    double averageVLevel = 0.0;
+    int vLevelSamples[numSamples];
+
+    for(int i = 0; i < numSamples; i++){
+        vLevelSamples[i] = analogRead(thermistorPin);
+        delay(10);
```

Comment:
At some point we'll have to re-factor this to not use delays because any delays in the main loop might cause us to miss serial data. For now we'll leave it as is, so we can get this merged but just something to be aware of.